core/io/resource_format_binary.cpp

```diff
@@ -2246,7 +2246,7 @@ Error ResourceFormatSaverBinaryInstance::save(const String &p_path, const Ref<Re
 				String new_id;
 
 				while (true) {
-					new_id = _resource_get_class(r) + "_" + Resource::generate_scene_unique_id();
+					new_id = _resource_get_class(r) + "_" + r->get_name() + "_" + itos(res_index);
```

Comment:
Know this is a draft but the loop isn't required here, and if for some reason the id is already used it will be stuck in an infinite loop here

Reply:
Ah yes, I'm still trying to figure out a better way, the loop maybe still necessary here to avoid duplication of name with external resources.

Is the number 5 fixed or not? Maybe I can do a hash on `new_id` and trim the hash.

Reply:
Generated here:
https://github.com/godotengine/godot/blob/4b7cc99bc195387c784efced7fdd360501d9a5cf/core/io/resource.cpp#L93-L121

Reply:
I have updated the code, use a similar compact form of hash. (The related code can be abstracted to another function).

I kind of ignored the empty name because I think type and sub resource order combined together maybe is enough for most cases.

Reply:
Will take a look perhaps later today!

Reply:
No hurry, take your time :)